sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -103,9 +105,11 @@ class SQLContext(@transient val sparkContext: SparkContext)
    */
   def getAllConfs: immutable.Map[String, String] = conf.getAllConfs
 
+  // TODO how to handle the temp table per user session?
```

Comment:
Yea, we can keep it as separated PR. But for Spark SQL, the temp table is managed by `Catalog`, probably we also need to refactor the `Catalog` code a little bit.